app/src/composables/use-relation/use-relation-multiple.ts

```diff
@@ -94,10 +94,15 @@ export function useRelationMultiple(
 			const updatedItem = cloneDeep(item);
 
 			if (editsIndex !== -1) {
-				merge(
+				mergeWith(
 					updatedItem,
 					{ $type: 'updated', $index: editsIndex, $edits: editsIndex },
-					_value.value.update[editsIndex]
+					_value.value.update[editsIndex],
+					(obj, src) => {
+						if (Array.isArray(obj) && Array.isArray(src)) {
```

Comment:
How does it handle JSON fields? Same problem?

Reply:
Tested on a JSON field:

## Before

https://user-images.githubusercontent.com/42867097/170508019-bba4154a-1b5c-419b-9d20-10b3b1b9854d.mp4

## After

https://user-images.githubusercontent.com/42867097/170508003-4847cd68-721b-41b1-a3ab-a6b76f20be5d.mp4

Reply:
@azrikahar What if the value goes from 

```json
{
  "value": {
    "a": "x"
  }
}
```

to

```json
{
  "value": {
    "b": "y" 
  }
}
```

do you end up with 

```json
{
  "value": {
    "b": "y" 
  }
}
```

or 

```json
{
  "value": {
    "a": "x",
    "b": "y" 
  }
}
```

?

Reply:
It was 100% turning into the latter as you suspected (rightfully so as it's also an expected behavior of `merge`):

https://user-images.githubusercontent.com/42867097/170643829-0712a974-06a1-4f4c-8473-b1a05430c717.mp4

That being said, I've updated the PR to treat edits as the single source of truth instead, and it does work for all the cases (including arrays) we've discussed:

https://user-images.githubusercontent.com/42867097/170643907-0e4da514-18dc-40fe-aecb-f9f2d245a070.mp4

Could there be any other edge cases not accounted for here? 🤔 Though once again this is technically the "display" being incorrect, but the actual update has been correct even prior to this PR:

https://user-images.githubusercontent.com/42867097/170644554-c8a58cde-f0ff-4c0a-9afc-14d15591bbc8.mp4

Reply:
Alternatively, relying on `assign` rather than `merge` might also do the trick. I think we'd only want to merge on the top-level properties here anyways right?